ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/CustomExecutors.java

```diff
@@ -0,0 +1,503 @@
+        final String statementText
+    ) {
+      final AggregateFunctionFactory aggregateFactory
+          = ksqlEngine.getMetaStore().getAggregateFactory(functionName);
+
+      final ImmutableList.Builder<FunctionInfo> listBuilder = ImmutableList.builder();
+
+      aggregateFactory.eachFunction(func -> listBuilder.add(
+          getFunctionInfo(func.getArgTypes(), func.getReturnType(), func.getDescription())));
+
+      return new FunctionDescriptionList(
+          statementText,
+          aggregateFactory.getName().toUpperCase(),
+          aggregateFactory.getDescription(),
+          aggregateFactory.getAuthor(),
+          aggregateFactory.getVersion(),
+          aggregateFactory.getPath(),
+          listBuilder.build(),
+          FunctionType.aggregate
+      );
+    }
+
+    private FunctionDescriptionList describeNonAggregateFunction(
+        final KsqlEngine ksqlEngine,
+        final String functionName,
+        final String statementText
+    ) {
+      final UdfFactory udfFactory = ksqlEngine.getMetaStore().getUdfFactory(functionName);
+
+      final ImmutableList.Builder<FunctionInfo> listBuilder = ImmutableList.builder();
+
+      udfFactory.eachFunction(func -> listBuilder.add(
+          getFunctionInfo(func.getArguments(), func.getReturnType(), func.getDescription())));
+
+      return new FunctionDescriptionList(
+          statementText,
+          udfFactory.getName().toUpperCase(),
+          udfFactory.getDescription(),
+          udfFactory.getAuthor(),
+          udfFactory.getVersion(),
+          udfFactory.getPath(),
+          listBuilder.build(),
+          FunctionType.scalar
+      );
+    }
+
+    private FunctionInfo getFunctionInfo(
+        final List<Schema> argTypes,
+        final Schema returnTypeSchema,
+        final String description
+    ) {
+      final List<ArgumentInfo> args = argTypes.stream()
+          .map(s -> new ArgumentInfo(s.name(), SchemaUtil.getSqlTypeName(s), s.doc()))
+          .collect(Collectors.toList());
+
+      final String returnType = SchemaUtil.getSqlTypeName(returnTypeSchema);
+
+      return new FunctionInfo(args, returnType, description);
+    }
+  },
+
+  SET_PROPERTY(SetProperty.class) {
+    @Override
+    public KsqlEntity execute(
+        final PreparedStatement statement,
+        final KsqlEngine ksqlEngine,
+        final ServiceContext serviceContext,
+        final KsqlConfig ksqlConfig,
+        final Map<String, Object> propertyOverrides
+    ) {
+      ksqlEngine.execute(statement, ksqlConfig, propertyOverrides);
+      return null;
+    }
+  },
+
+  UNSET_PROPERTY(UnsetProperty.class) {
+    @Override
+    public KsqlEntity execute(
+        final PreparedStatement statement,
+        final KsqlEngine ksqlEngine,
+        final ServiceContext serviceContext,
+        final KsqlConfig ksqlConfig,
+        final Map<String, Object> propertyOverrides
+    ) {
+      ksqlEngine.execute(statement, ksqlConfig, propertyOverrides);
+      return null;
```

Comment:
change the interface to return `Optional<KsqlEntity>` rather than returning nulls, i.e. use the type system to indicate the method optionally returns a value.